soc/arm/nxp_imx/mimx8ml8_m7/soc.c

```diff
@@ -321,22 +106,19 @@ static void SOC_ClockInit(void)
 	/* switch AHB to SYSTEM PLL1 DIV6 */
 	CLOCK_SetRootMux(kCLOCK_RootAhb, kCLOCK_AhbRootmuxSysPll1Div6);
 
-	/* Set root clock freq to 800MHZ/ 2= 400MHZ*/
-	CLOCK_SetRootDivider(kCLOCK_RootAudioAhb, 1U, 2U);
-	/* switch AUDIO AHB to SYSTEM PLL1 */
-	CLOCK_SetRootMux(kCLOCK_RootAudioAhb, kCLOCK_AudioAhbRootmuxSysPll1);
-
+#ifdef CONFIG_UART_MCUX_IUART
```

Comment:
Since it looks like this clock is specific to UART4, you may want to add to this `#ifdef` to check that the uart4 node is enabled in the DTS. See here for an example: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_imx/rt/soc_rt10xx.c#L198

Reply:
done! See latest update.